batch/batch/worker/worker.py

```diff
@@ -474,17 +487,27 @@ async def localize_rootfs():
             self.state = 'error'
             self.error = traceback.format_exc()
         finally:
-            with self.step('deleting'):
-                await self.delete_container()
+            await self.delete_container()
```

Comment:
This is partially a separate issue that we need to address in a followup. The step manager will raise on `__enter__` if the job has been deleted, which means it will double raise and if we time anything in the `finally` block. This is part of the reason why job cancellation wasn't working properly. In the future I think the step manager should not care about job deletion since we now have an `asyncio.Event` to signal deletion. I'm unsure whether it's better to leave it in this in between and clean up deletion handling later or lump more things that don't seem entirely related into this PR, since many things use the timings blocks.